nextjs/src/data/puck/config/blocks/FilterableGrid/index.tsx

```diff
@@ -125,6 +145,32 @@ export const FilterableGrid: ComponentConfig<FilterableGridProps> = {
                     },
                     timestamp: {
                         type: "number"
+                    },
+                    eventMonth: {
+                        // @ts-ignore
+                        visible: data.props.type === "event",
+                        type: "select",
+                        options: eventMonths
+                    },
+                    eventDay: {
+                        // @ts-ignore
+                        visible: data.props.type === "event",
+                        type: "text",
+                    },
+                    eventTime: {
+                        // @ts-ignore
+                        visible: data.props.type === "event",
+                        type: "text",
+                    },
+                    eventLocation: {
+                        // @ts-ignore
+                        visible: data.props.type === "event",
+                        type: "text",
+                    },
+                    imageUrl: {
+                        // @ts-ignore
+                        visible: data.props.type === "resource",
+                        type: "text"
```

Comment:
`visible` isn't a thing in Puck and this logic doesn't do anything. The fields are defined for *all* items in the data. We'd need to use separate drag n drop blocks to have conditional logic on a per-card basis.

A real limitation of Puck I'm afraid.

Reply:
Nonetheless it works fine and looks good.

Reply:
ahh ok, I thought that might be the case.